packages/qwik/src/core/util/string.ts

```diff
@@ -0,0 +1,128 @@
+// pack bytes into valid UTF-16 string
+//
+// strategy:
+//
+// * using esc as the escape character
+//  * if there is esc in the bytes, double it
+// * if there is unmatched surrogate pair, mark it by the escape character
+//
+// 0x007f: escape, because it's rare but still only one utf-8 byte.
+//   To escape itself, use 0x007f 0x08ff (two bytes utf-8)
+// 0x0000->0x001f: converted to esc + 0x0020->0x003f (two bytes utf-8)
+// unmatched pairs: converted to esc + (code-0xd800+0x0040) 0x0040->0x083f
+//   (two-four bytes utf-8)
+// BOM: esc + 0x08fe (four bytes utf-8)
+//
+// If the length of the bytes is odd, the last byte XX is put after the escape
+// character as 0xFFXX.
+//
+const ESC = 0x007f;
+const SURROGATE_OFFSET = 0xd800 - 0x0040;
+export const packUint8Array = (bytes: Uint8Array) => {
+  const odd = bytes.length % 2 === 1;
+  const dbytes = new Uint16Array(bytes.buffer, 0, bytes.length >> 1);
```

Comment:
Actually, I think you have to use the bytes offset of the u8 array here?

Could be problematic when it starts at an odd location :-/

Definitely worth adding some tests.